tests/foreman/cli/test_realm.py

```diff
@@ -0,0 +1,316 @@
+                        'realm-type': 'Red Hat Identity Management'})
+
+    @tier1
+    def test_negative_create_invalid_realm_type(self):
+        """Create a realm with an invalid type
+
+        :id: 423a0969-9311-48d2-9220-040a42159a89
+
+        :expectedresults: Realm creation fails, type must be in list
+            e.g. Red Hat Identity Management or Active Directory
+        """
+        with self.assertRaises(CLIFactoryError):
+            make_realm({'name': 'test', 'realm-proxy-id': '1',
+                        'realm-type': 'invalid_type'})
+
+    @tier1
+    def test_negative_create_invalid_location(self):
+        """Create a realm with an invalid location
+
+        :id: 95335c3a-413f-4156-b727-91b525738171
+
+        :expectedresults: Realm creation fails, location not found
+
+        """
+        with self.assertRaises(CLIFactoryError):
+            make_realm({'name': 'test', 'realm-proxy-id': '1',
+                        'locations': 'Raleigh, NC',
+                        'realm-type': 'invalid_type'})
+
+    @tier1
+    def test_negative_create_invalid_organization(self):
+        """Create a realm with an invalid type
+
+        :id: c0ffbc6d-a2da-484b-9627-5454687a3abb
+
+        :expectedresults: Realm creation fails, organization not found
+        """
+        with self.assertRaises(CLIFactoryError):
+            make_realm({'name': 'test', 'realm-proxy-id': '1',
+                        'organizations': 'Red Hat',
+                        'realm-type': 'invalid_type'})
+
+    @tier1
+    def test_negative_delete_nonexistent_realm_name(self):
+        """Delete a realm with a name that does not exist
+
+        :id: 616db509-9643-4817-ba6b-f05cdb1cecb0
+
+        :expectedresults: Realm not found
+        """
+        with self.assertRaises(CLIReturnCodeError):
+            Realm.delete({'name': 'dne'})
+
+    @tier1
+    def test_negative_delete_nonexistent_realm_id(self):
+        """Delete a realm with an id that does not exist
+
+        :id: 70bb9d4e-7e71-479a-8c82-e6fcff88ea14
+
+        :expectedresults: Realm not found
+        """
+        with self.assertRaises(CLIReturnCodeError):
+            Realm.delete({'id': 0})
+
+    @tier1
+    def test_negative_info_nonexistent_realm_name(self):
+        """Get info for a realm with a name that does not exist
+
+        :id: 24e4fbfa-7141-4f90-8c5d-eb88b162bd64
+
+        :expectedresults: Realm not found
+        """
+        with self.assertRaises(CLIReturnCodeError):
+            Realm.info({'name': 'dne'})
+
+    @tier1
+    def test_negative_info_nonexistent_realm_id(self):
+        """Get info for a realm with an id that does not exists
+
+        :id: db8382eb-6d0b-4d6a-a9bf-38a462389f7b
+
+        :expectedresults: Realm not found
+        """
+        with self.assertRaises(CLIReturnCodeError):
+            Realm.info({'id': 0})
+
+    @tier1
+    def test_negative_update_nonexistent_realm_name(self):
```

Comment:
I would say that both info and update test cases look as pretty redundant as you actually verify that entity is not found rather than you cannot update it (like when entity can be only numerical and you try to update it with wrong type)